bed/bedPe/bedPe.go

```diff
@@ -0,0 +1,185 @@
+	switch {
+	case fields == 6:
+		return fmt.Sprintf("%s\t%d\t%d\t%s\t%d\t%d", bunk.Chrom1, bunk.ChromStart1, bunk.ChromEnd1, bunk.Chrom2, bunk.ChromStart2, bunk.ChromEnd2)
+	case fields == 7:
+		return fmt.Sprintf("%s\t%d\t%d\t%s\t%d\t%d\t%s", bunk.Chrom1, bunk.ChromStart1, bunk.ChromEnd1, bunk.Chrom2, bunk.ChromStart2, bunk.ChromEnd2, bunk.Name)
+	case fields == 8:
+		return fmt.Sprintf("%s\t%d\t%d\t%s\t%d\t%d\t%s\t%d", bunk.Chrom1, bunk.ChromStart1, bunk.ChromEnd1, bunk.Chrom2, bunk.ChromStart2, bunk.ChromEnd2, bunk.Name, bunk.Score)
+	case fields == 9:
+		return fmt.Sprintf("%s\t%d\t%d\t%s\t%d\t%d\t%s\t%d\t%c", bunk.Chrom1, bunk.ChromStart1, bunk.ChromEnd1, bunk.Chrom2, bunk.ChromStart2, bunk.ChromEnd2, bunk.Name, bunk.Score, bunk.Strand1)
+	case fields == 10:
+		return fmt.Sprintf("%s\t%d\t%d\t%s\t%d\t%d\t%s\t%d\t%c\t%c", bunk.Chrom1, bunk.ChromStart1, bunk.ChromEnd1, bunk.Chrom2, bunk.ChromStart2, bunk.ChromEnd2, bunk.Name, bunk.Score, bunk.Strand1, bunk.Strand2)
+	case fields >= 11:
+		var out string = fmt.Sprintf("%s\t%d\t%d\t%s\t%d\t%d\t%s\t%d\t%c\t%c", bunk.Chrom1, bunk.ChromStart1, bunk.ChromEnd1, bunk.Chrom2, bunk.ChromStart2, bunk.ChromEnd2, bunk.Name, bunk.Score, bunk.Strand1, bunk.Strand2)
+		for i := range bunk.Annotation {
+			out = fmt.Sprintf("%s\t%s", out, bunk.Annotation[i])
+		}
+		return out
+	default:
+		log.Fatalf("Error: expecting a request to print at least 6 bedPe fields, but got: %d\n", fields)
+	}
+	return ""
+}
+
+// WriteBed writes an input BedPe struct to an io.Writer.
+func WriteBedPe(file io.Writer, input BedPe) {
+	var err error
+	_, err = fmt.Fprintf(file, "%s\n", input)
+	exception.PanicOnErr(err)
+}
+
+// WriteToFileHandle writes an input BedPe struct to an io.Writer
+func WriteToFileHandle(file io.Writer, rec BedPe) {
+	var err error
+	_, err = fmt.Fprintf(file, "%s\n", rec)
+	exception.PanicOnErr(err)
+}
+
+//Write writes a slice of BedPe structs to a specified filename.
+func Write(filename string, records []BedPe) {
+	var err error
+	file := fileio.EasyCreate(filename)
+
+	for i := range records {
+		WriteToFileHandle(file, records[i])
+	}
+	err = file.Close()
+	exception.PanicOnErr(err)
+}
+
+// Read returns a slice of BedPe structs from an input filename.
+func Read(filename string) []BedPe {
+	var line string
+	var answer []BedPe
+	var err error
+	var doneReading bool
+
+	file := fileio.EasyOpen(filename)
+
+	for line, doneReading = fileio.EasyNextRealLine(file); !doneReading; line, doneReading = fileio.EasyNextRealLine(file) {
+		current := processBedPeLine(line)
+		answer = append(answer, current)
+	}
+	err = file.Close()
+	exception.PanicOnErr(err)
+	return answer
+}
+
+//processBedPeLine is a helper function of Read that returns a BedPe struct from an input line of a file.
+func processBedPeLine(line string) BedPe {
+	var err error
+	var start1Num, end1Num, start2Num, end2Num int
+	words := strings.Split(line, "\t")
+	start1Num, err = strconv.Atoi(words[1])
```

Comment:
Nothing wrong with this, but I think we have functions that do this conversion and check.  Maybe in "numbers" or "fileio"?